ledger/heavy/proc/send_pulse_test.go

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright 2019 Insolar Technologies GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package proc_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/gojuno/minimock/v3"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/insolar/insolar/insolar"
+	"github.com/insolar/insolar/insolar/bus"
+	"github.com/insolar/insolar/insolar/payload"
+	"github.com/insolar/insolar/insolar/pulse"
+	"github.com/insolar/insolar/instrumentation/inslogger"
+	"github.com/insolar/insolar/ledger/heavy/proc"
+)
+
+func TestSendPulse_Proceed(t *testing.T) {
+	mc := minimock.NewController(t)
+	ctx := inslogger.TestContext(t)
+
+	var (
+		sender *bus.SenderMock
+		pulses *pulse.AccessorMock
+	)
+
+	resetComponents := func() {
+		sender = bus.NewSenderMock(mc)
+		pulses = pulse.NewAccessorMock(mc)
+	}
+
+	newProc := func(msg payload.Meta) *proc.SendPulse {
+		p := proc.NewSendPulse(msg)
+		p.Dep(pulses, sender)
+		return p
+	}
+
+	resetComponents()
+	t.Run("pulse not found", func(t *testing.T) {
+		p := newProc(payload.Meta{})
+		pulses.ForPulseNumberMock.Return(insolar.Pulse{}, pulse.ErrNotFound)
+
+		err := p.Proceed(ctx)
+		require.Error(t, err)
+		assert.Equal(t, pulse.ErrNotFound, errors.Cause(err))
+
+		mc.Finish()
```

Comment:
Let's call it right after `setup` as `defer`.

Reply:
done